From: Zhang Rui <rui.zhang@intel.com>
Date: Wed, 22 Apr 2020 15:26:07 +0800
Subject: ACPI: processor: idle: Allow probing on platforms with one ACPI
 C-state
Patch-mainline: v5.8-rc1
Git-commit: 496121c02127e9c460b436244c38260b044cc45a
References: jsc#SLE-16407

It is possible for ACPI _CST to return only one ACPI C-state, for
example, when deep cstate disabled in the BIOS.

And it is better for the acpi_idle driver to probe succesfully in
this case as well for consistency.

Signed-off-by: Zhang Rui <rui.zhang@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 drivers/acpi/processor_idle.c |    9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

--- a/drivers/acpi/processor_idle.c
+++ b/drivers/acpi/processor_idle.c
@@ -308,11 +308,7 @@ static int acpi_processor_get_power_info
 	if (ret)
 		return ret;
 
-	/*
-	 * It is expected that there will be at least 2 states, C1 and
-	 * something else (C2 or C3), so fail if that is not the case.
-	 */
-	if (pr->power.count < 2)
+	if (!pr->power.count)
 		return -EFAULT;
 
 	pr->flags.has_cst = 1;
@@ -468,8 +464,7 @@ static int acpi_processor_get_cstate_inf
 	for (i = 1; i < ACPI_PROCESSOR_MAX_POWER; i++) {
 		if (pr->power.states[i].valid) {
 			pr->power.count = i;
-			if (pr->power.states[i].type >= ACPI_STATE_C2)
-				pr->flags.power = 1;
+			pr->flags.power = 1;
 		}
 	}
 
